lib/compiler.js

```diff
@@ -0,0 +1,129 @@
+'use strict';
+
+/**
+ * See spec: https://www.w3.org/TR/webvtt1/#file-structure
+ */
+
+function CompilerError (message, error) {
+  this.message = message;
+  this.error = error;
+}
+CompilerError.prototype = Object.create(Error.prototype);
+
+function compile (input) {
+
+  if (!input) {
+    throw new CompilerError('Input must be non-null');
+  }
```

Comment:
I want spaces between blocks. eslint rules weren't catching but I just updated `master`, adding the `padding-line-between-statements`